### DISCUSSION TITLE
Host git repo on own server

### DISCUSSION THREAD

**Discussion by EleanorPrins** [community/community#23635] [New to GitHub]
<p>Hello, would there be any way to host a git repo on my own debian 10 machine? Meaning when you navigate to the repo on the website you can see the local files on my machine, and any push and pull just takes the files from the server. Is this possible using github?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That sounds like a situation where you specifically wouldn’t use Github, and instead set up a git hosting server on your own machine. Git comes with an <a href="https://git-scm.com/book/en/v2/Git-on-the-Server-GitWeb">integrated web interface called GitWeb</a>, to try it out quickly see <a href="https://git-scm.com/docs/git-instaweb"><code>git instaweb</code></a>.</p>
<p>There are a bunch of other self-hosting options if GitWeb doesn’t have enough features for you, but I’m not sure if there are any others that can work directly from your working repository.</p>

**Comment by EleanorPrins** [comment]
<p>I don’t think this is what i’m looking for at the moment, but thanks for the suggestion anyways.</p>
